### ISSUE TITLE
2020-008 - (SERB) - SMDS and CMDS Board Members Template Header Block

### ISSUE THREAD

**Issue by anthonyperk** [CurrentlyNotSet/SERB#736]
A new board member was recently appointed to SERB to replace the prior Vice Chairman. The new board member has the designation as Board Member and we do not currently have a Vice Chair. I am unable to make this change in the database and have it reflect on the template headers.

Can we update the database and necessary bookmark codes “HeaderPersonnelBlock” so that we can specify the Title change, and order of three board members presented (1,2,3), in the header code? I believe the title is currently hard coded.

It is possible that a Vice Chair could be determined as early as February 27th…. But this scenario will repeat iteself in the future. We wanted to reach out to determine how best to approach this scenario and the amount of time it will take you to implement an update.

We would want the same tweak implemented on the SPBR side.

With Vice Chair Example:
![image](https://user-images.githubusercontent.com/24595073/74046439-931b7f00-499c-11ea-97af-8fb73dc31599.png)

Without Vice Chair Example:
![image](https://user-images.githubusercontent.com/24595073/74046467-9d3d7d80-499c-11ea-946f-7b8068df04e0.png)

![image](https://user-images.githubusercontent.com/24595073/74046479-a0d10480-499c-11ea-93dd-75e697f046b9.png)

**Comment by anthonyperk** [comment]
Clarification from Justin:

> Anthony, there will only ever be up to three board members. However their titles can change.
> Examples:
> Chairman, Board Member, Board Member
> Chair, Vice Chair, Board Member
> Chairman, Vice Chairman, Board Member
> Chairwoman, Vice Chairwoman, Board Member
> Etc.

**Comment by altrdgenetics** [comment]
The Code to load up the personnel block can be found here:

![Screenshot (35)](https://user-images.githubusercontent.com/8847344/74206696-7f954000-4c4a-11ea-9626-6b7be4671cc3.png)
![Screenshot (34)](https://user-images.githubusercontent.com/8847344/74206700-82903080-4c4a-11ea-9d14-413694eaa453.png)


**The following changes will need to occur in order to facilitate this request:**

- We will need to add a numbers field to the database 
- We will need to adjust the admin screens to support the selection of the personnel block. In order to do that I think we should come up with a table that allows the admin to basically drag and drop on a table to "create" the block of their choice
- all of this code needs to be removed and we will need to re-grab everything and iterate over all of the "selected" persons
- from the screens above the fields in the database for the "titles" are aliased to what is in the bookmarks. We will need to make sure this gets changed over since as of right now those are basically hard coded in.

NOTE: the table will be a little extra work but will be able to add more flexibility to SERB admin to create their own personnel block and allow for "drag and drop" ordering. The fallback alternative would be to allow the user to freeform put in an order number (int value). The risk here would be the admin could put in 2 of the same numbers. This would be because they would have to go into each line item in the admin panel to input the selected number, making that change tedious.

**Comment by anthonyperk** [comment]
EMAIL FROM JUSTIN

> I spoke with Christine and Erin. The “Vice Chair” appointment should be voted on Feb 27th so we do not need to rush any coding to solve our situation. However, we do want to have the coding in place for the future. 
> 
> C**an you proceed with 2020-008 by providing a quote for your recommended solution with the drag and drop for both the SERB and SPBR Boards**
>

**Comment by anthonyperk** [comment]
So SERB is looking for an estimate of hours along with a mock-up (digital or hand written/drawn) of the window as well as how it would work.

**Comment by anthonyperk** [comment]
MORE INFO

Estimate of hours. A basic mockup (hand drawn would even be OK) to visualize the solution.

If the board members and Exec Director were listed in the database as follows… That would be OK too.
We really only need to change the name and the job title. We could reenter them into the system if the order needed changed. 

1.	Craig Zimpher, Chairman, Phone, Email
2.	Fred Mills, Board Member, Phone, Email
3.	Richard Lumpe, Board Member, Phone, Email
4.	Christine Dietsch, Executive Director, Phone, Email

**Comment by altrdgenetics** [comment]
Quick mockup. There is one issue I see after looking closer at the templates and the code. There is a space between the board an Chris as Executive Director... I am not entirely sure how to handle that within this flow...

I might be able to streamline the arrows or maybe allow for drag and drop within the same table for order. But that would all depend on if I can solve that quickly.

![Screenshot (39)](https://user-images.githubusercontent.com/8847344/75614809-80631880-5b0a-11ea-92c4-47cd9d078b75.png)

**Comment by anthonyperk** [comment]
EMAIL FROM JUSTIN

Please proceed with the work as you outlined below. At this time we are only concerned with the SERB side

Also:
Chief Administrator Law Judge position is not needed
Yes, one blank line between the Board Member and Executive Director

**Comment by altrdgenetics** [comment]
The bookmark order itself is accessable from the "SERB Board" Dialog stored within the Admin panel.

The Bookmark itself is a direct inline replacement for "HeaderPersonnelBlock".

**Comment by anthonyperk** [comment]
Testing Results:
------------------------------------
1) There is no space in between the board members and the Executive Director

2) Justin also requested this for the PBR side as well.

Everything else was great

**Comment by anthonyperk** [comment]
FROM JUSTIN

Yes we want the blank Line between Board members and Christine’s name. If we are short a board member we would want two blank rows.

**Comment by altrdgenetics** [comment]
Here is how it functions now. If they want I can expand blank lines based on Board titles. 

This is now being dynamically built out based on a numbering system (And Active). Adding in BLANKS is not really possible unless they want to add in "dummy blank" entries into the database.


![Screenshot (116)](https://user-images.githubusercontent.com/8847344/84792680-27889480-afc2-11ea-9da1-d7732c855ecd.png)

**Comment by altrdgenetics** [comment]
OLD bookmark setup for creating the personnel block

![Screenshot (117)](https://user-images.githubusercontent.com/8847344/84796844-1f7f2380-afc7-11ea-9443-e2aed924ca28.png)

**Comment by anthonyperk** [comment]
EMAIL FROM JUSTIN

I thoroughly tested ticket 2020-008 (Board Members Header Bookmark). This ticket addressed the issue we had with Fred Mills being a Board Member until voted to Vice Chair. The new build works very well. It is clean, crisp, and easy to use. I spent 15 minutes trying to break it. Below are the only two issues that I was able to find, when throwing different scenarios at it that might occur in the future. 

1)	The system changes the position title chairman to chair. But it keeps chairwomen and chairperson with no issues. I think there is some code overriding the use of the word chairman.  
2)	If we only have a two member board, we want the ability to add in a blank spot so that the Executive Director position does not float up from its current position like in the example below. 
![image](https://user-images.githubusercontent.com/24595073/85051841-0660a900-b166-11ea-92e7-b7654198b566.png)


I tested a wide variety of possible board member titles, reordered them, etc. and I really like how the system implemented the changes via the bookmark.

**Comment by anthonyperk** [comment]
TESTING RESULTS

1) Good.  I can change the title to whatever I want and it prints it out exactly as I entered it in.  I even changed it to Chair"Person" and thats what it printed.

2) I tried to add a 'blank' member, but when it printed, the 'space' in between the last board member and the executive director didnt print, though the blank member did 'print'.  I had to add a second blank member to get the formatting correct, however with the blank board members, the comma between name and title still prints.
